src/coroutine.rs

```diff
@@ -135,3 +158,21 @@ impl Coroutine {
         self.poll(py, None)
     }
 }
+
+impl Drop for Coroutine {
+    fn drop(&mut self) {
+        if self.future.is_some() {
+            Python::with_gil(|gil| {
```

Comment:
This structure reminds me of https://github.com/PyO3/pyo3/discussions/3386#discussioncomment-6732913.

I'm not entirely sure if it's a blocker to not have `try_with_gil` or some other guard to avoid potential nasty crashes here. We should have a little think about that before merge.

Reply:
Ahh that's a good suggestion, I should really pick up `__del__` support again soon. Are you suggesting split this PR in two; merge the names now and keep the warning as a draft pr for the moment?

Reply:
I am indeed.

Reply:
I've opened an issue for the warning.